identity_stardust/examples/create_did.rs

```diff
@@ -28,79 +35,46 @@ use identity_stardust::StardustDID;
 use identity_stardust::StardustDocument;
 use identity_stardust::StardustVerificationMethod;
 
-// PROBLEMS SO FAR:
-// 1) Alias Id is inferred from the block, so we have to use a placeholder DID for creation.
-// 2) Cannot get an Output Id back from an Alias Id (hash of Output Id), need to use Indexer API.
-// 3) The Output response from the Indexer is an Output, not a Block, so cannot infer Alias ID from it (fine since we
-// use the ID to retrieve the Output in the first place).    The OutputDto conversion is annoying too.
-// 4) The pieces needed to publish an update are fragmented (Output ID for input, amount, document), bit annoying to
-// reconstruct.    Use a holder struct like Holder { AliasOutput, StardustDocument } with convenience functions?
-// 5) Inferred fields such as the controller and governor need to reflect in the (JSON) Document but excluded from the
-// StardustDocument serialization when published.    Handle with a separate `pack` function like before?
+static ENDPOINT: &str = "https://api.testnet.shimmer.network/";
+static FAUCET_URL: &str = "https://faucet.testnet.shimmer.network/api/enqueue";
```

Comment:
I guess we want to get away from predefining node endpoints, at least on the main and shimmer network (i.e. not test networks), so probably not?
I'm still in favor of retrieving the HRP at runtime as already partially done in this PR.

Reply:
> I guess we want to get away from predefining node endpoints

Do we? I think at least having them hardcoded in the library for convenience (whether in a `Network` or just as constants) would be useful.
And again the `HRP` can conflict between networks/chains, so I don't think it's sufficient as a network identifier.

Reply:
> Do we?

I'd lean towards it, yes, to encourage people to run their own nodes and not use IF nodes.

> And again the HRP can conflict between networks/chains

Yeah, although it only seems to be a problem for the test networks not for main or shimmer net. So I'm not sure how much of a problem it really is in practice. E.g. when you want to resolve and `rms` DID you don't know which network it is so it's unclear what endpoint to use. Again, though, only an issue in test scenarios from my understanding.

We can alternatively use the `network_id` of the network (a `u64`) which should be unique, and hardcode mappings to our own names like `main`, `smr`, `smr_testnet`, whatever. Then it'll be unambiguous what network to resolve a `did:iota:smr:...` on.

Reply:
> I'd lean towards it, yes, to encourage people to run their own nodes and not use IF nodes.

What about users that just want to resolve identities?

> it only seems to be a problem for the test networks not for main or shimmer net

It seems like a problem for anyone using a private Tangle too, or forks, or potentially shards, or whatever else could be done in the future out of our control.

> We can alternatively use the `network_id` of the network

How stable is that, does it change between network updates? Will it change after Stardust hits the IOTA Mainnet?

Reply:
> What about users that just want to resolve identities?

I honestly don't know if we want to encourage resolvers to just trust IF nodes. 

> How stable is that

Not as stable as I hoped:

> Network identifier. This field denotes whether the message was meant for mainnet, testnet, or a private net. It also marks what protocol rules apply to the message. Usually, it will be set to the first 8 bytes of the BLAKE2b-256 hash of the concatenation of the network type and the protocol version string.

https://github.com/iotaledger/tips/blob/f4e1daa2b830695a848f437de30a8ddfe25f4f45/tips/TIP-0006/tip-0006.md#serialized-layout

Reply:
> I honestly don't know if we want to encourage resolvers to just trust IF nodes.

Not maintaining a default list of endpoints just seems like a big barrier to entry for anyone looking to start out using IOTA Identity. Even regular IOTA users might only be familiar with what Firefly handles automatically for them.

> Not as stable as I hoped:

Unfortunate, thanks for looking up the information. It would be a problem if that network identifier changes and breaks backwards compatibility arbitrarily, so maybe not the way forward.

Might need to look at what other DID methods do again, I know some use explicit network identifiers but we should see.

Reply:
> Not maintaining a default list of endpoints just seems like a big barrier to entry

That is fair. I would leave adding a `Network` type for another PR, though.